service-registry/src/main/java/io/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -89,6 +89,10 @@ protected boolean doRegister() {
 
     private void checkSchemaIdSet() {
         Microservice existMicroservice = srClient.getMicroservice(microservice.getServiceId());
+        if (existMicroservice == null) {
```

Comment:
yes, maybe is not the properly way to just return. I found this problem when the connection to Service Center is not stable.  sdk is failed to update heartbeat to SC or any time when sdk register to SC. but if the register thread exits, it will not cover any more even next time sdk could update heartbeat successfully ,it's an improvement issue,